test/test_regression_cli.py

```diff
@@ -1,34 +1,54 @@
-from __future__ import print_function
+import imp
 import os
-from unittest import TestCase
-from hashlib import md5
 
-import nose
+import pytest
 
+from retriever.lib.tools import getmd5
+from retriever import HOME_DIR, ENGINE_LIST
 
-# First md5 is for csv, second md5 is for sqlite
-known_md5s = {
+mysql_engine, postgres_engine, sqlite_engine, msaccess_engine, csv_engine, download_engine, json_engine, xml_engine = ENGINE_LIST()
+dir_path = os.path.dirname(os.path.abspath(__file__))
 
-    'sqlite': {'AvianBodySize': '72256f681cdce96eba32d4ece270bcb2',
-               'DelMoral2010': 'fe0adb120f33479357f1b9673d5883c2',
-               'MoM2003': 'd3bdce86e0fc5888449884dfb0ef4611'},
+download_md5 = [
+    # ('DelMoral2010', '0'),
+    ('avianbodysize', 'dce81ee0f040295cd14c857c18cc3f7e'),
+    ('mom2003', 'b54b80d0d1959bdea0bb8a59b70fa871')
+]
 
-    'download': {'AvianBodySize': 'dce81ee0f040295cd14c857c18cc3f7e',
-                 'DelMoral2010': '2b6e92b014ae73ea1f0195ecfdf6248d',
-                 'MoM2003': 'b54b80d0d1959bdea0bb8a59b70fa871'},
+csv_md5 = [
+    # ('DelMoral2010', '0'),
+    ('EA_avianbodysize2007', '873597e93bb6163aedd88e285b246c34'),
+    ('EA_mom2003', '5fe8b4f182ba2b8be3bb0ddecdb4673c')
+]
 
-    'csv': {'AvianBodySize': 'f42702a53e7d99d16e909676f30e5aa8',
-            'DelMoral2010': 'e79d55ac15f1a70a6c7d3ad4e678ec0e',
-            'MoM2003': 'ef0a31c132cfe1c6594739c872f70f54'},
+db_md5 = [
+    # ('DelMoral2010', '0'),
+    ('EA_avianbodysize2007', '79680888f7768474479e70c87cd36c9d'),
+    ('EA_mom2003', '520b29f78dc96e8f2d48b5ef9264ff86')
+]
 
-    'mysql': {'AvianBodySize': 'f60ac93d9be4671dbef77da9d10676b8',
-              'DelMoral2010': 'f241fd296130512d4e1029376b58a4ea',
-              'MoM2003': '9728728d72af4c21a2a6e29fec3edb48'},
+filedb_md5 = [
+    # ('DelMoral2010', '0'),
+    ('EA_avianbodysize2007', '91aa87a438ddf8fc6c9f9054fe4fda90'),
+    ('EA_mom2003', '8dc11ac8d3d3a33e938662dfe39e22c2')
+]
 
-    'postgres': {'AvianBodySize': '60c252af74d914e3c15fa9af43edefca',
-                 'DelMoral2010': '63d91a972b5ca07bf90900a576286a8a',
-                 'MoM2003': 'a55c8308722c8e20950e0d1e6d9639e6'}
-}
+
+def get_script_module(script_name):
+    """Load a script module"""
+    file, pathname, desc = imp.find_module(script_name, [os.path.join(HOME_DIR, "scripts")])
+    return imp.load_module(script_name, file, pathname, desc)
+
+
+def dump_csv(dataset, engines):
```

Comment:
Since this function finishes by calculating the md5 we should name it something reflects that like `get_csv_md5`. Alternatively we could move the last two lines into the `test_` functions and leave this named as it is.